### ISSUE TITLE
Can we save BERT fine tuned model created by text.text_classifier( ), but without the last classification layer?

### ISSUE THREAD

**Issue by hebatallah** [amaiya/ktrain#95]
Hello,

Can we save text classification BERT fine tuned model without the last classification layer?. In order to be reused by other classification tasks (with different output labels).

I think we can´t do that if we used:
model = text.text_classifier('bert', train_data=(x_train, y_train), preproc=preproc),
instead we must use:
model = text.Transformer(...)

Right?

Thanks

**Comment by amaiya** [comment]
v0.12.x has a `TransformerEmbedding` class that can be used to generate word or sentence embeddings for downstream tasks with pretrained (vs. fine-tuned) weights. These are the pretrained model without classification layer (not fine-tuned model).  If you're looking to save the fine-tuned embeddings after training, you can take a look at [this thread](https://github.com/huggingface/transformers/issues/1001)

**Comment by amaiya** [comment]
Once you save a model, you can reload it for other purposes (E.g., generate embeddings):

```python
# load text data
categories = ['alt.atheism', 'soc.religion.christian','comp.graphics', 'sci.med']
from sklearn.datasets import fetch_20newsgroups
train_b = fetch_20newsgroups(subset='train', categories=categories, shuffle=True)
test_b = fetch_20newsgroups(subset='test',categories=categories, shuffle=True)
(x_train, y_train) = (train_b.data, train_b.target)
(x_test, y_test) = (test_b.data, test_b.target)

# build, train, and validate model (Transformer is wrapper around transformers library)
import ktrain
from ktrain import text
MODEL_NAME = 'distilbert-base-uncased'
t = text.Transformer(MODEL_NAME, maxlen=500, class_names=train_b.target_names)
trn = t.preprocess_train(x_train, y_train)
val = t.preprocess_test(x_test, y_test)
model = t.get_classifier()
learner = ktrain.get_learner(model, train_data=trn, val_data=val, batch_size=6)
learner.fit_onecycle(5e-5, 1)

# load model to generate embeddingsmodel.save_pretrained('/tmp/mymodel')
from transformers import *
import tensorflow as tf
tokenizer = AutoTokenizer.from_pretrained(MODEL_NAME)
model = TFBertModel.from_pretrained('/tmp/mymodel')
input_ids = tf.constant(tokenizer.encode("Hello, my dog is cute"))[None, :]  # Batch size 1
outputs = model(input_ids)
last_hidden_states = outputs[0]  # The last hidden-state is the first element of the output tuple
print(last_hidden_states.numpy().shape)
```
